File: searchnsort/ltcd_611.py
'''
LeetCode - problem 611
'''

    def triangleNumber(self, nums: List[int]) -> int:
        nums=sorted(nums)
        c=0
        for i in range(len(nums)-1,-1,-1):
            j,k=0,i-1
            while j<k:
                if nums[j]+nums[k]>nums[i]:
                    c+=k-j
                    k-=1
                else:
                     j+=1
        return c

File: searchnsort/ltcd_611.cpp
//LeetCode - problem 611

    int triangleNumber(vector<int>& nums) {
        int cnt = 0;
        int n = nums.size();
        sort(nums.begin(),nums.end());
        for(int i=2;i<n;i++)
        {
            int l = 0;
            int h = i-1;
            while(l<h)
            {
                if(nums[l]+nums[h] > nums[i])
                {
                    cnt += (h-l);
                    h--;
                }
                else
                {
                    l++;
                }
            }
        }
        return cnt;
    }
